src/main/resources/graphql/schema.graphqls

```diff
@@ -0,0 +1,122 @@
+scalar Long
+
+type Query {
+    getGalaxies(name: String, pageInput: PageInput = {}): [GalaxyDto]
+    getGalaxy(id: ID!): GalaxyDto
+
+    getStars(name: String, pageInput: PageInput = {}): [StarDto]
+    getStar(id: ID!): StarDto
+
+    getPlanets(name: String, pageInput: PageInput = {}): [PlanetDto]
+    getPlanet(id: ID!): PlanetDto
+
+    getMoons(name: String, pageInput: PageInput = {}): [MoonDto]
+    getMoon(id: ID!): MoonDto
+}
+
+type Mutation {
+    addGalaxy(input: AddGalaxyInput!) : GalaxyDto
+    updateGalaxy(input: UpdateGalaxyInput!) : GalaxyDto
+    deleteGalaxy(id: ID!): ID
+
+    addStar(input: AddStarInput!) : StarDto
+    updateStar(input: UpdateStarInput!) : StarDto
+    deleteStar(id: ID!): ID
+
+    addPlanet(input: AddPlanetInput!) : PlanetDto
+    updatePlanet(input: UpdatePlanetInput!) : PlanetDto
+    deletePlanet(id: ID!): ID
+
+    addMoon(input: AddMoonInput!) : MoonDto
+    updateMoon(input: UpdateMoonInput!) : MoonDto
+    deleteMoon(id: ID!): ID
+}
+
+input PageInput {
+    page: Int = 0
+    size: Int = 20
+    sort: [SortOrder] = []
+}
+
+input SortOrder {
+    property: String!
+    direction: Direction = ASC
+}
+
+enum Direction {
```

Comment:
This enum should be renamed to `SortDirection`, as we might have other direction types.

Reply:
It's named "Direction" intentionally, to comply with Sort.Direction. Otherwise new enum should be created on java side.

Reply:
No need for new java enum. Should work as is after you rename it to `SortDirection`.